generated/nidcpower/nidcpower/session.py

```diff
@@ -4822,11 +4822,28 @@ def fetch_multiple(self, count, timeout=hightime.timedelta(seconds=1.0)):
 
         '''
         import collections
-        Measurement = collections.namedtuple('Measurement', ['voltage', 'current', 'in_compliance'])
-
-        voltage_measurements, current_measurements, in_compliance = self._fetch_multiple(timeout, count)
-
-        return [Measurement(voltage=voltage_measurements[i], current=current_measurements[i], in_compliance=in_compliance[i]) for i in range(count)]
+        Measurement = collections.namedtuple('Measurement', ['voltage', 'current', 'in_compliance', 'channel'])
+
+        voltage_measurements, current_measurements, in_compliances = self._fetch_multiple(timeout, count)
+
+        with _NoChannel(session=self):
+            all_channels_in_session = self._get_channel_names(range(self.channel_count))
```

Comment:
Add a comment here:

```
#TODO(olsl21): retrieving the list of channels in the session on every function call is silly because they never change #12345
```

Where 12345 is the issue number for fixing this.

Reply:
I have created #1776 and added the TODO comment in my latest commit.